cmake/modules/extensions.cmake

```diff
@@ -2268,6 +2269,22 @@ Please provide one of following: APPLICATION_ROOT, CONF_FILES")
       endif()
     endforeach()
 
+    if(SYSBUILD)
+      foreach(path ${${FILE_APPLICATION_ROOT}})
```

Comment:
afaict this will adjust any relative path to absolute path when current Zephyr build is invoked by sysbuild.

What we should do instead is to use `zephyr_file()` from within sysbuild itself to adjust any relative paths given by the user as `-D` before writing sysbuild cache.

For example here: https://github.com/zephyrproject-rtos/zephyr/blob/a17fc4ff2b71cc833ae479c2aaf87d014fd87a5b/share/sysbuild/cmake/modules/sysbuild_extensions.cmake#L319
where any `*_ROOT` can be adjusted by `zephyr_file()` (where zephyr_file needs to adjust according to `APP_DIR` fot the common root.

The image specific relative root, such as `mcuboot_*_ROOT` can also be adjusted here, or postponed till image specific call to `zephyr_file`, similar to your current proposal. And then cache the image specific sysbuild var locally, similar to this:
https://github.com/zephyrproject-rtos/zephyr/blob/a17fc4ff2b71cc833ae479c2aaf87d014fd87a5b/cmake/modules/extensions.cmake#L2256-L2260

so that only cached and image specific sysbuild vars are updated.

Note, if doing it later, then you must be take out only the sysbuild image specific form of `<image>_<VAR>_ROOT`, which is not posiible today.

Reply:
Updated code